lib/quicly.c

```diff
@@ -4395,7 +4399,10 @@ int quicly_receive(quicly_conn_t *conn, struct sockaddr *dest_addr, struct socka
     switch (conn->super.state) {
     case QUICLY_STATE_CLOSING:
         conn->super.state = QUICLY_STATE_DRAINING;
```

Comment:
Why does this state change to DRAINING immediately? This should only be happening when a CONNECTION_CLOSE is received, right?

Reply:
Nice catch. Yes, we need to remove this line.

As we do not decrypt packets once entering the CLOSING state, we need to stay there for three PTO.

Reply:
So, we'll never be in DRAINING?

Reply:
We enter DRAINING when we *receive* close.

Reply:
Fixed in da20776.